### ISSUE TITLE
Several accessibility issues

### ISSUE THREAD

**Issue by R-J** [vanilla/vanilla#9009]
Based on a blog post, I searched for tools helping evaluating the accessibility of a web page and came up with [this one](http://wave.webaim.org/report#/https://open.vanillaforums.com/discussion/37360/vanilla-3-0-is-now-available-updated-3-0-2#latest).

The main takeaways are the following:
- The RichEditor embeddings need to have an "alt" attribute
- The AdvancedSearch box has several issues and might need some reconsiderations
- Having `echo userAnchor, userPhoto` results in username repetition because the images alt attribute [wouldn't be needed here](https://webaim.org/standards/wcag/checklist#sc1.1.1)
- The social links in the footer need (sr-only) texts

I was surprised about that superfluous alt text. I think this could be solved quite elegantly by adding something like `$altText = $options['Alt'] ?? $name` to the function userPhoto, so that it can be used `echo userPhoto($Author, ['Alt' => '']);

**Comment by charrondev** [comment]
We have some plans right now for alt text on rich editor embeddings. You can track that issue here https://github.com/vanilla/vanilla/issues/8507 and it's planned for the next release.

> The AdvancedSearch box has several issues and might need some reconsiderations

It might need elaboration. For our knowledge product we've made a new search box that should be a lot better, and I'd like to bring that back to the forum rather than touch the existing forum stuff.

This likely will get done when we bring the new search page to the forum side. There's still a lot of details to hash out with that.

> Having echo userAnchor, userPhoto results in username repetition because the images alt attribute wouldn't be needed here

This one is deserving of it's own issue I think. It should be a pretty easy fix.

> The social links in the footer need (sr-only) texts

Which social links? Our default theme don't have any social links in the footer.

![image](https://user-images.githubusercontent.com/1770056/60816752-5e2a1e00-a168-11e9-847e-be3605db31ae.png)
